### PR TITLE
feat: MOFED subscription mounts

### PR DESCRIPTION
**Pull request by rollandf**
Add needed mounts for subscription in RHEL with non CRIO container runtime.

### CODE REVIEW

File: pkg/nodeinfo/node_info.go [hunk 1]
```diff
@@ -127,6 +129,8 @@ func (p *provider) GetNodePools(filters ...Filter) []NodePool {
 		}
 		nodePool.Kernel = kernel
 
+		nodePool.ContainerRuntime = getContainerRuntime(node)
```
Comment (adrianchiris) [inline pkg/nodeinfo/node_info.go:132]: do you assume that container runtime is the same for all nodes that are part of the pool ?

technically every attribute that is not part of the key in nodePoolMap may be different and you just save the last node that match.

File: pkg/state/state_ofed.go [hunk 2]
```diff
@@ -783,6 +825,40 @@ func (s *stateOFED) handleCertConfig(
 	return nil
 }
 
+// handleSubscriptionVolumes handles additional mounts required for subscriptions
+func (s *stateOFED) handleSubscriptionVolumes(
+	ctx context.Context, osname string, runtime string, mounts *additionalVolumeMounts) error {
+	reqLogger := log.FromContext(ctx)
+	if osname == "rhel" && runtime != nodeinfo.CRIO {
```
Comment (adrianchiris) [inline pkg/state/state_ofed.go:832]: do we want to ignore an empty runtime value ?

Comment (rollandf) [reply pkg/state/state_ofed.go:832]: Can an empty runtime happen?

**Comment by rollandf** [comment]
Tested manually on cluster with RHEL 8.8, containerd 1.7.16
MOFED pods compiles successfully

File: pkg/state/state_ofed.go (hunk 2 above)
Comment (adrianchiris) [reply pkg/state/state_ofed.go:832]: field should be set. just wondering if we should add check for completness.

File: pkg/nodeinfo/node_info.go (hunk 1 above)
Comment (adrianchiris) [reply pkg/nodeinfo/node_info.go:118]: ack.

**Review by adrianchiris — APPROVED** [review]

**Pull request merged**